src/server/scripts/Pandaria/zone_the_wandering_isle.cpp

```diff
@@ -0,0 +1,162 @@
+/*
+ * This file is part of the TrinityCore Project. See AUTHORS file for Copyright information
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "Player.h"
+#include "ScriptMgr.h"
+#include "MotionMaster.h"
+#include "ScriptedCreature.h"
+#include "TaskScheduler.h"
+
+enum Aspirant
+{
+    SAY_FINISH_FIGHT = 0,
+    SPELL_BLACKOUT_KICK = 109080,
+    QUEST_29524_KILLCREDIT = 54586,
+    EVENT_RANDOM_EMOTE = 1,
+    EVENT_INSTRUCTOR_ZHI_RANDOM_EMOTE = 2,
+    NPC_INSTRUCTOR_ZHI = 61411,
+    ACTION = 1
+};
+
+Emote randomEmotes[5] =
+{
+    EMOTE_ONESHOT_MONKOFFENSE_ATTACKUNARMED,
+    EMOTE_ONESHOT_MONKOFFENSE_SPECIALUNARMED,
+    EMOTE_ONESHOT_MONKOFFENSE_PARRYUNARMED,
+    EMOTE_ONESHOT_PALMSTRIKE,
+    EMOTE_ONESHOT_MONKOFFENSE_ATTACKUNARMEDOFF,
+};
+
+//54586
+struct npc_aspiring_trainee : public ScriptedAI
+{
+    npc_aspiring_trainee(Creature* c) : ScriptedAI(c)
+    {
+        events.ScheduleEvent(EVENT_INSTRUCTOR_ZHI_RANDOM_EMOTE, 6s);
+    }
+
+    void Reset() override
```

Comment:
this part can be removed, if you are not using anything besides `ScriptedAI::Reset()` its implicitly called (since we are inheriting from ScriptedAI)